### PR TITLE
Clean up optional and inconsistently named inputs [VS-294] [VS-218]

### PR DESCRIPTION
**Pull request by rsasch**
- hard-code inputs that shouldn't be override-able
- name inputs consistently

Closes
- https://broadworkbench.atlassian.net/browse/VS-294
- https://broadworkbench.atlassian.net/browse/VS-218

### CODE REVIEW

File: scripts/variantstore/TERRA_QUICKSTART.md [hunk 1]
```diff
@@ -1,125 +1,106 @@
...
 These are the required parameters which must be supplied to the workflow:
 
 | Parameter             | Description |
 | --------------------- | ----------- |
-| project_id            | The name of the google project containing the dataset |
-| dataset_name          | The name of the dataset you created above       |
-| external_sample_names | datamodel  (e.g `this.samples.sample_id`)     |
+| dataset_name          | the name of the dataset you created above       |
+| external_sample_names | `this.samples.sample_id` (the sample identifier column from the `gvs_demo_10` sample set) |
+| project_id            | the name of the google project containing the dataset |
```
Comment (RoriCremer) [inline scripts/variantstore/TERRA_QUICKSTART.md:34]: omg is it alpha now?!??!?!

File: scripts/variantstore/TERRA_QUICKSTART.md [hunk 2]
```diff
@@ -1,125 +1,106 @@
...
-This step extracts the data in BigQuery into a sharded joint called VCF 
-
-To prepare the dataset for extraction by putting it into a single vet and ref_ranges table, run the `GvsPrepareRangesCallset` workflow with the following parameters:
+| Parameter                         | Description |
+| --------------------------------- | ----------- |
+| dataset_name                      | the name of the dataset you created above  |
+| filter_set_name                   | a unique name to identify this filter set (e.g. `my_demo_filters` ); you will want to make note of this for use in step 5 |
+| INDEL_VQSR_max_gaussians_override | you don't need to set this unless a previous run of IndelsVariantRecalibrator task failed to converge, start with 3 and lower as needed |
+| project_id                        | the name of the google project containing the dataset |
+| SNP_VQSR_max_gaussians_override   | you don't need to set this unless a previous run of SNPsVariantRecalibratorClassic task failed to converge, start with 5 and lower as needed |
```
Comment (RoriCremer) [inline scripts/variantstore/TERRA_QUICKSTART.md:76]: this is awesome!!!!

**Review by RoriCremer — APPROVED** [review]

File: scripts/variantstore/wdl/GvsAssignIds.wdl [hunk 3]
```diff
@@ -26,24 +26,24 @@ workflow GvsAssignIds {
       dataset_name = dataset_name,
       datatype = "sample_info",
       schema_json = sample_info_schema_json,
-      max_table_id=1,
+      max_table_id = 1,
       superpartitioned = "false",
       partitioned = "false",
       service_account_json_path = service_account_json_path,
-      preemptible_tries = preemptible_tries,
+      preemptible_tries = create_tables_preemptible_override,
```
Comment (kcibul) [inline scripts/variantstore/wdl/GvsAssignIds.wdl:33]: ToL: for these very short, small VM tasks that alter the database should we just use "real" VMs and not have the PE option?  Then we also don't have to worry about getting pre-empted and handling that.

File: scripts/variantstore/wdl/GvsCreateFilterSet.wdl [hunk 4]
```diff
@@ -4,605 +4,553 @@ import "GvsWarpTasks.wdl" as Tasks
...
+
+  # this is the minimum number of samples where the SNP model will be created and applied in separate tasks
+  # (SNPsVariantRecalibratorClassic vs. SNPsVariantRecalibratorCreateModel and SNPsVariantRecalibratorScattered)
+  Int snps_variant_recalibration_threshold = 20000
+
+  File gatk_override = "gs://broad-dsp-spec-ops/scratch/bigquery-jointcalling/jars/kc_extract_perf_20220111/gatk-package-4.2.0.0-455-g40a40bc-SNAPSHOT-local.jar"
+
+  Array[String] snp_recalibration_tranche_values = ["100.0", "99.95", "99.9", "99.8", "99.6", "99.5", "99.4", "99.3", "99.0", "98.0", "97.0", "90.0" ]
+
+  # reference files
```
Comment (kcibul) [inline scripts/variantstore/wdl/GvsCreateFilterSet.wdl:32]: So happy to see this!!

File: scripts/variantstore/wdl/GvsCreateFilterSet.wdl [hunk 5]
```diff
@@ -4,605 +4,553 @@ import "GvsWarpTasks.wdl" as Tasks
...
+    call ExtractFilterTask {
+      input:
+        gatk_override            = gatk_override,
+        reference                = reference,
+        reference_index          = reference_index,
+        reference_dict           = reference_dict,
+        fq_sample_table          = fq_sample_table,
+        intervals                = SplitIntervals.interval_files[i],
+        fq_alt_allele_table      = fq_alt_allele_table,
+        excess_alleles_threshold = 1000000,
```
Comment (kcibul) [inline scripts/variantstore/wdl/GvsCreateFilterSet.wdl:90]: we should make a ticket to just remove this "feature" from the filters… it didn't work out.  Maybe we need to confirm with Lee, I think it was his idea?

File: scripts/variantstore/wdl/GvsCreateFilterSet.wdl [hunk 6]
```diff
@@ -4,605 +4,553 @@ import "GvsWarpTasks.wdl" as Tasks
...
+      recalibration_tranche_values = ["100.0", "99.95", "99.9", "99.5", "99.0", "97.0", "96.0", "95.0", "94.0", "93.5", "93.0", "92.0", "91.0", "90.0"],
+      recalibration_annotation_values = indel_recalibration_annotation_values,
+      mills_resource_vcf = mills_resource_vcf,
+      mills_resource_vcf_index = mills_resource_vcf_index,
+      axiomPoly_resource_vcf = axiomPoly_resource_vcf,
+      axiomPoly_resource_vcf_index = axiomPoly_resource_vcf_index,
+      dbsnp_resource_vcf = dbsnp_vcf,
+      dbsnp_resource_vcf_index = dbsnp_vcf_index,
+      use_allele_specific_annotations = true,
+      disk_size = "1000",
```
Comment (kcibul) [inline scripts/variantstore/wdl/GvsCreateFilterSet.wdl:120]: no longer configurable?  haven't we needed to tweak this in the past along with machine memory size?

File: scripts/variantstore/wdl/GvsCreateFilterSet.wdl [hunk 7]
```diff
@@ -4,605 +4,553 @@ import "GvsWarpTasks.wdl" as Tasks
...
+    gatk --java-options "-Xmx4g" ExtractFeatures \
+      --ref-version 38  \
+      -R "~{reference}" \
+      -O "~{output_file}" \
+      --local-sort-max-records-in-ram 1000000 \
+      --sample-table ~{fq_sample_table} \
+      --alt-allele-table ~{fq_alt_allele_table} \
+      ~{"--excess-alleles-threshold " + excess_alleles_threshold} \
+      -L ~{intervals} \
+      --project-id ~{read_project_id}
```
Comment (kcibul) [inline scripts/variantstore/wdl/GvsCreateFilterSet.wdl:345]: I think we're losing something here

File: scripts/variantstore/wdl/GvsCreateTables.wdl [hunk 8]
```diff
@@ -2,47 +2,50 @@ version 1.0
...
-      String vetDone = CreateVetTables.done
-      String refDone = CreateRefRangesTables.done
-    }
+  input {
+    String dataset_name
+    String project_id
+
+    Int max_table_id
+
+    Int? preemptible_tries
```
Comment (kcibul) [inline scripts/variantstore/wdl/GvsCreateTables.wdl:11]: Same ToL as above… maybe we shouldn't even do this with PE VMs… the benefit of PE is cost, and the downside is that we could get pre-empted in the middle of doing something and retry

File: scripts/variantstore/wdl/GvsExtractCallset.wdl [hunk 9]
```diff
@@ -3,295 +3,249 @@ version 1.0
...
+
+  Array[String] tables_patterns_for_datetime_check = ["~{extract_table_prefix}__%"]
+
+  call Utils.SplitIntervals {
+    input:
+      intervals = interval_list,
+      ref_fasta = reference,
+      ref_fai = reference_index,
+      ref_dict = reference_dict,
+      interval_weights_bed = "gs://broad-public-datasets/gvs/weights/gvs_vet_weights_1kb.bed",
```
Comment (kcibul) [inline scripts/variantstore/wdl/GvsExtractCallset.wdl:47]: Can this still be overridden?

File: scripts/variantstore/wdl/GvsAssignIds.wdl (hunk 3 above)
Comment (rsasch) [reply scripts/variantstore/wdl/GvsAssignIds.wdl:33]: I'm happy to remove it 👍🏻

File: scripts/variantstore/wdl/GvsCreateFilterSet.wdl (hunk 6 above)
Comment (rsasch) [reply scripts/variantstore/wdl/GvsCreateFilterSet.wdl:120]: I have only had to tweak the memory size in my runs.  I needed to tweak the disk size for `SplitIntervals`, tho.

**Pull request merged**